### PR TITLE
Refs [TASK] [36055] Show Collections and Photo Collection

### PR DESCRIPTION
**Pull request by vvanan158**
## Related Tickets
- [#Task 36055:  Show Collections and Photo Collection](https://edu-redmine.sun-asterisk.vn/issues/36055)
## WHAT
-  Show list collections and photo Collections
## Evidence (Screenshot or Video)
image
image

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/unsplash_01/data/repository/CollectionRepository.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    suspend fun getCollections(page: Int = DEFAULT_PAGE) = withContext(Dispatchers.IO) {
+        return@withContext client.getCollections(page = page)
+    }
+
+    suspend fun getPhotosCollection(
+        id: String,
+        page: Int = DEFAULT_PAGE
+    ) = withContext(Dispatchers.IO) {
+        return@withContext client.getPhotosCollection(id = id, page = page)
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/unsplash_01/data/repository/CollectionRepository.kt]: coroutine chạy dưới viewModel. sao lại đem lên đến đây e?

File: app/src/main/java/com/sun/unsplash_01/ui/collection/AdapterCollection.kt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.sun.unsplash_01.R
+import com.sun.unsplash_01.data.model.Collection
+import com.sun.unsplash_01.databinding.ItemLayoutCollectionBinding
+import com.sun.unsplash_01.utils.OnItemClickListener
+
+class AdapterCollection : ListAdapter<Collection, CollectionViewHolder>(Companion) {
+
+    private var onClick: ((Collection) -> Unit)? = null
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/unsplash_01/ui/collection/AdapterCollection.kt]: đặt tên thể hiện chức năng click cái gì?

File: app/src/main/java/com/sun/unsplash_01/ui/collection/AdapterCollection.kt [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        fun from(parent: ViewGroup, onClick: ((Collection) -> Unit)): CollectionViewHolder {
+            val binding = DataBindingUtil.inflate<ItemLayoutCollectionBinding>(
+                LayoutInflater.from(parent.context),
+                R.layout.item_layout_collection, parent,
+                false
+            )
+            return CollectionViewHolder(binding, onClick).apply {
+                registerOnItemClickListener()
+            }
+        }
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/unsplash_01/ui/collection/AdapterCollection.kt]: sao lại đưa vào companion vậy e?

File: app/src/main/java/com/sun/unsplash_01/ui/collection/CollectionFragment.kt [hunk 4]
```diff
@@ -1,14 +1,23 @@
...
+import com.sun.unsplash_01.R
 import com.sun.unsplash_01.databinding.FragmentCollectionBinding
+import com.sun.unsplash_01.utils.Status
+import org.koin.android.ext.android.inject
 
 class CollectionFragment : Fragment() {
 
+    private val collectionViewModel: CollectionViewModel by inject()
+    private val collectionAdapter by lazy { AdapterCollection() }
     lateinit var binding: FragmentCollectionBinding
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/unsplash_01/ui/collection/CollectionFragment.kt]: hôm qua mới presentation convention

File: app/src/main/java/com/sun/unsplash_01/ui/collection/CollectionFragment.kt [hunk 5]
```diff
@@ -19,7 +28,51 @@ class CollectionFragment : Fragment() {
         return binding.root
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding()
+        registerObserver()
+        handleEvent()
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/unsplash_01/ui/collection/CollectionFragment.kt]:
```suggestion
        handleEvent()
        registerObserver()
```

File: app/src/main/java/com/sun/unsplash_01/ui/photo_collection/AdapterPhotoCollection.kt [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+import androidx.recyclerview.widget.RecyclerView
+import com.sun.unsplash_01.R
+import com.sun.unsplash_01.data.model.Collection
+import com.sun.unsplash_01.data.model.PhotoCollection
+import com.sun.unsplash_01.databinding.ItemLayoutPhotoBinding
+import com.sun.unsplash_01.utils.OnItemClickListener
+
+class AdapterPhotoCollection : ListAdapter<PhotoCollection, PhotoCollectionViewHolder>(Companion) {
+
+    private var onClick: ((PhotoCollection) -> Unit)? = null
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/unsplash_01/ui/photo_collection/AdapterPhotoCollection.kt]: đặt lại tên

File: app/src/main/java/com/sun/unsplash_01/ui/photo_collection/AdapterPhotoCollection.kt [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+            val binding = DataBindingUtil.inflate<ItemLayoutPhotoBinding>(
+                LayoutInflater.from(parent.context),
+                R.layout.item_layout_photo, parent,
+                false
+            )
+            return PhotoCollectionViewHolder(binding, onClick).apply {
+                registerOnItemClickListener()
+            }
+        }
+    }
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/unsplash_01/ui/photo_collection/AdapterPhotoCollection.kt]: as above

File: app/src/main/java/com/sun/unsplash_01/ui/photo_collection/PhotoCollectionFragment.kt [hunk 8]
```diff
@@ -5,10 +5,17 @@ import android.view.LayoutInflater
...
 import com.sun.unsplash_01.databinding.FragmentPhotoCollectionBinding
+import com.sun.unsplash_01.ui.collection.CollectionFragment.Companion.BUNDLE_COLLECTION
+import com.sun.unsplash_01.utils.Status
+import org.koin.android.ext.android.inject
 
 class PhotoCollectionFragment : Fragment() {
 
+    private val photoCollectionViewModel: PhotoCollectionViewModel by inject()
+    private val photoCollectionAdapter by lazy { AdapterPhotoCollection() }
     lateinit var binding: FragmentPhotoCollectionBinding
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/unsplash_01/ui/photo_collection/PhotoCollectionFragment.kt]: sắp xếp

File: app/src/main/java/com/sun/unsplash_01/ui/collection/AdapterCollection.kt (hunk 3 above)
Comment (vvanan158) [reply app/src/main/java/com/sun/unsplash_01/ui/collection/AdapterCollection.kt]: dạ e tách code ra lun đấy a

File: app/src/main/java/com/sun/unsplash_01/data/repository/CollectionRepository.kt (hunk 1 above)
Comment (vvanan158) [reply app/src/main/java/com/sun/unsplash_01/data/repository/CollectionRepository.kt:18]: e muốn await kết quả trả về để set isload lại a

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**